negotiation/sessions/neural_session.py

```diff
@@ -269,6 +271,14 @@ def _is_valid(self, tokens):
         return True
 
     def output_to_tokens(self, data):
-        pred = data["predictions"][0][0]
-        entity_tokens = self.builder._build_target_tokens(pred)
-        return entity_tokens
+        predictions = data["predictions"][0][0]
+        clean_tokens = []
+        for pred in predictions:
+            token = self.vocab.ind_to_word[pred]
+            if is_entity(token):
+                token = self.dialogue.entity_to_price(self.kb, token)
+            clean_tokens.append(token)
+            if clean_tokens[-1] == markers.EOS:
+                clean_tokens = clean_tokens[:-1]
+                break
+        return clean_tokens
```

Comment:
can we put this in _build_target_token?

Reply:
I tried calling self.builder(pred) at one point, but needed too many hacks to make it work, mostly because the function needs access to self.dialogue and self.kb which utterance builder doesn't hase.  Also, the function needs to extract the predictions from the data which is a different step.

Reply:
can we add `kb` as a param with default value `None` in `_build_target_token`, so that here you can call it with the additional input `kb`?

Reply:
I updated session so that it calls on _build_target_tokens.  I feel like the set-up is a bit messy, but maybe there's a cleaner way to write this.   Let me know what you prefer.